### PR TITLE
Consolidate all `App` code into `Command`

### LINKED ISSUE #1586: Consolidate all `App` code into `Command`
The `App` struct and `Command` struct have very similar surface area and behaviors. Greatly reducing the duplication between them has already been done in the `v2` series, but I'd like to see if a single `Command` struct with `.Commands` field can be used everywhere and the `App` struct can be removed (or type alias?).

### LINKED ISSUE #1600: Use `.Commands` field throughout (replace `.Subcommands`)

### PR DESCRIPTION
**Pull request by meatballhat**
## What type of PR is this?

_(REQUIRED)_

- bug
- cleanup
- documentation
- feature

## What this PR does / why we need it:

_(REQUIRED)_

## Which issue(s) this PR fixes:

_(REQUIRED)_

Fixes #1586 

## Special notes for your reviewer:

_(fill-in or delete this section)_

## Testing

_(fill-in or delete this section)_

## Release Notes

_(REQUIRED)_

```release-note

```

### CODE REVIEW

**Comment by dearchap** [comment]
@meatballhat Perhaps its better to split this into two separate PRs. One to change Subcommands -> Commands and another to merge App into command.

**Comment by meatballhat** [comment]
> @meatballhat Perhaps its better to split this into two separate PRs. One to change Subcommands -> Commands and another to merge App into command.

@dearchap that was my intention when I started, but keeping both `.Commands` and `.Subcommands` didn't seem feasible 🤔 Are you suggesting I take on the `.Subcommands` ➡️ `.Commands` rename _first_?

**Comment by dearchap** [comment]
@meatballhat One thing that may help is removing ctx from Help and using Command params throughout. That way there are no App/Ctx references anywhere on that portion. If you'd like I can work on that.

**Comment by meatballhat** [comment]
> @meatballhat One thing that may help is removing ctx from Help and using Command params throughout. That way there are no App/Ctx references anywhere on that portion. If you'd like I can work on that. 

@dearchap that sounds great to me! :tada:

File: help_test.go [hunk 1]
```diff
@@ -574,58 +574,6 @@ func TestShowCommandHelp_CommandAliases(t *testing.T) {
 	}
 }
 
-func TestHelpNameConsistency(t *testing.T) {
```
Comment (dearchap) [inline help_test.go:577]: @meatballhat Any reason why this was removed ?

Comment (meatballhat) [reply help_test.go:577]: Oh gosh I totally don't remember why I did this 😭 I'm starting over!

Comment (meatballhat) [reply help_test.go:577]: Ooh, I remember! But I'm still starting over 😂 I'll be introducing smaller PRs to do parts of what's discovered here.

**Comment by meatballhat** [comment]
Superseded by #1746

**Pull request closed without merging**